test/integration/targets/rds_instance/tasks/test_snapshot.yml

```diff
@@ -0,0 +1,24 @@
+---
+  - block:
+      - name: set up aws connection info
+        set_fact:
+          aws_connection_info: &aws_connection_info
+            aws_access_key: "{{ aws_access_key }}"
+            aws_secret_key: "{{ aws_secret_key }}"
+            security_token: "{{ security_token }}"
+            region: "{{ aws_region }}"
+        no_log: yes
+
+      - name: Getting shared snapshots
+        rds_snapshot_info:
+          snapshot_type: "shared"
+          <<: *aws_connection_info
+        register: result
+
+      - assert:
+          that:
+            - not result.changed
+            - result.cluster_snapshots is defined
+            - result.snapshots is defined
+
+      # TODO ideally we test with an actual shared snapshot - but we'd need a second account - making tests fairly complicated?
```

Comment:
I agree, this would need quite a bit of work to support.  We can at least test that the code path doesn't have any obvious errors though so thanks very much for the test.